src/main/java/server/Router.java

```diff
@@ -40,30 +41,33 @@ private void addToRoutes(HTTPMethods method, String path, Handler handler) {
     }
 
     public Response generateResponse(Request request) {
+        String path = request.getRequestPath();
+        if(path.contains("?")){
+            path = path.split("\\?")[0];
+        }
 
-        if (!pathExists(request)) {
-
+        if (!pathExists(path)) {
             return ResponseTypes.notFound(request);
         }
 
-        if (!methodAllowed(request)) {
+        if (!methodAllowed(request, path)) {
             HashMap<String, Handler> pathMethods = routes.get(request.getRequestPath());
```

Comment:
can probably ✂️ some calls to `request.getRequestPath()` now that we've got a `path` var?